src/SqlAsyncCollector.cs

```diff
@@ -379,7 +392,9 @@ private static void GenerateDataQueryForMerge(TableInformation table, IEnumerabl
 
         public class TableInformation
         {
-            public IEnumerable<PropertyInfo> PrimaryKeys { get; }
+            public List<PrimaryKey> PrimaryKeys { get; }
```

Comment:
It would be nice to follow this up with merging these into a single Tuple list - avoids confusion about why we'd need to have two separate lists that both contain a value for each primary key.